controllers/component_controller.go

```diff
@@ -47,10 +60,197 @@ type ComponentReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := r.Log.WithValues("Component", req.NamespacedName)
+	log.Info(fmt.Sprintf("Starting reconcile loop for %v", req.NamespacedName))
+
+	// Fetch the Component instance
+	var hasComponent appstudiov1alpha1.Component
+	err := r.Get(ctx, req.NamespacedName, &hasComponent)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// If the devfile hasn't been populated, the CR was just created
+	if hasComponent.Status.Devfile == "" {
+		source := hasComponent.Spec.Source
+		context := hasComponent.Spec.Context
+		var devfilePath string
+
+		// append context to devfile if present
+		// context is usually set when the git repo is a multi-component repo (example - contains both frontend & backend)
+		if context == "" {
+			devfilePath = devfileName
+		} else {
+			devfilePath = path.Join(context, devfileName)
+		}
+
+		if source.GitSource != nil && source.GitSource.URL != "" {
+			var devfileBytes []byte
+
+			if source.GitSource.DevfileURL == "" {
+				rawURL, err := util.ConvertGitHubURL(source.GitSource.URL)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to convert Github URL to raw format, exiting reconcile loop %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+
+				endpoint := rawURL + "/" + devfilePath
+				devfileBytes, err = util.CurlEndpoint(endpoint)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to curl %s, attempting to clone the repo and read the devfile instead %v", devfilePath, req.NamespacedName))
+
+					// clone the repo and read the devfile
+					clonePath := path.Join(clonePathPrefix, hasComponent.Spec.Application, hasComponent.Spec.ComponentName)
+					devfileBytes, err = util.CloneAndReadDevfile(clonePath, devfilePath, source.GitSource.URL)
```

Comment:
Do we clean up and remove the temp folder after we clone and retrieve the devfile? If not, we should.